scipy/stats/stats.py

```diff
@@ -5967,9 +5973,13 @@ def ttest_ind(a, b, axis=0, equal_var=True, nan_policy='propagate',
     if a.size == 0 or b.size == 0:
         return _ttest_nans(a, b, axis, Ttest_indResult)
 
-    if permutations:
-        if int(permutations) != permutations or permutations < 0:
-            raise ValueError("Permutations must be a positive integer.")
+    if permutations is not None:
+        if permutations < 0 or (np.isfinite(permutations) and
+                                int(permutations) != permutations) :
+            raise ValueError("Permutations must be a non-negative integer.")
+
+        if permutations == 0:
+            permutations = np.inf
```

Comment:
I think we should only have this behavior for the `np.inf` case-that may mean this branches needs to change.

Reply:
Yup.